stablehlo/dialect/StablehloOps.cpp

```diff
@@ -4660,48 +4660,80 @@ LogicalResult ReplicaIdOp::inferReturnTypes(
 // If Op
 //===----------------------------------------------------------------------===//
 
-static LogicalResult verifyConditionalBranch(Operation* op, Region& region,
-                                             llvm::Twine branchName) {
+static LogicalResult verifyConditionalBranch(Region& region,
+                                             llvm::Twine branchName,
+                                             Optional<Location> loc) {
   if (region.getNumArguments() != 0)
-    return op->emitOpError()
-           << branchName << " must have 0 arguments, but found "
-           << region.getNumArguments();
-
-  TypeRange branchReturnTypes =
-      region.front().getTerminator()->getOperandTypes();
-  if (branchReturnTypes != op->getResultTypes())
-    return op->emitOpError()
-           << branchName << " returned types (" << branchReturnTypes
-           << ") do not match op result types (" << op->getResultTypes() << ")";
-
+    return emitOptionalError(loc, branchName,
+                             " must have 0 arguments, but found ",
+                             region.getNumArguments());
   return success();
 }
 
-LogicalResult IfOp::verify() {
-  if (failed(verifyConditionalBranch(*this, true_branch(),
-                                     /*branchName=*/"true_branch"))) {
-    return failure();
+bool isCompatibleForTypeRange(ValueTypeRange<OperandRange> typeRange1,
+                              ValueTypeRange<OperandRange> typeRange2) {
+  if (typeRange1.size() != typeRange2.size()) return false;
+  for (auto it : llvm::zip(typeRange1, typeRange2)) {
+    if (!mlir::hlo::isCompatibleForHloTypeInference(std::get<0>(it),
+                                                    std::get<1>(it)))
+      return false;
   }
+  return true;
+}
```

Comment:
I think that having a function that takes `TypeRange` will be sufficient, because `TypeRange` has constructor that takes `ValueTypeRange`.

Reply:
Done.